### PR TITLE
Ft pagination and filtering 153051694

### PR DESCRIPTION
**Pull request by mirr254**
**What does this PR do**

Implements pagination, filtering events by location and sets limits per page

**Tasks completed**

- Pagination implementation
- implement events filtering by location 
- searching event by name

**Related stories**
[153051694](https://www.pivotaltracker.com/story/show/153051694)

**Screenshots**
location-filter

page-limit

### CODE REVIEW

File: app/events_blueprint/views.py [hunk 1]
```diff
@@ -77,7 +77,8 @@ def get_all_events(logged_in_user):
     if check_blacklisted_token(token) == True:
         return jsonify({'Session expired':'Please login again'}),403
     page = request.args.get('page',1,type=int )
-    events = models.Events.query.paginate(page, app.config['POSTS_PER_PAGE'], False).items
+    limit = request.args.get('limit',3,type=int) #defaults to 3 if user doesn't specify to limit
```
Comment (frankip) [inline app/events_blueprint/views.py]: nice implementation

File: app/events_blueprint/views.py [hunk 2]
```diff
@@ -134,15 +135,16 @@ def edit_event(logged_in_user,eventid):
     return response
 
 #search by name
-@events.route('/api/v1/events/<string:location>', methods=['GET'])
+@events.route('/api/v1/events/location/', methods=['GET'])
 @token_required
```
Comment (frankip) [inline app/events_blueprint/views.py]: do you really require a token to search? I think anybody should be able to search regardless of whether they are logged in or not

File: app/events_blueprint/views.py [hunk 3]
```diff
@@ -135,15 +135,15 @@ def edit_event(logged_in_user,eventid):
...
 def searc_by_location(logged_in_user):
      #check if token is blacklisted
     token = request.headers['x-access-token']
     if check_blacklisted_token(token) == True:
         return jsonify({'Session expired':'Please login again'}),403
-    location = session.args.get('location')
-    event_searched = models.Events.query.filter( models.Events.location.like('%'+location+'%'))
+    name = request.args.get('q')
+    event_searched = models.Events.query.filter( models.Events.name.like('%'+name+'%'))
```
Comment (frankip) [inline app/events_blueprint/views.py]: good implementation

File: app/events_blueprint/views.py [hunk 4]
```diff
@@ -135,15 +135,15 @@ def edit_event(logged_in_user,eventid):
...
     import pdb; pdb.set_trace()
```
Comment (frankip) [inline app/events_blueprint/views.py]: what does this pdb and pdb.set_trace() do?

**Review by frankip — COMMENTED** [review]
good implementation, but my concern is that anybody should be able to filter and search for events without having to signing in

**Comment by mirr254** [comment]
@frankip okay. I hadn't put that in mind

**Comment by mirr254** [comment]
I think you are right

**Pull request merged**